client/pkid_client.go

```diff
@@ -0,0 +1,253 @@
+package client
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	sodium "github.com/gokillers/libsodium-go/cryptosign"
+)
+
+type PkidClient struct {
+	client     http.Client
+	port       int
+	privateKey []byte
+	publicKey  []byte
+}
+
+// create a new instance from the pkid client
+func NewPkidClient(privateKey []byte, publicKey []byte, port int) PkidClient {
+	client := http.Client{Timeout: 5 * time.Second}
```

Comment:
maybe we can have the timeout configured in the NewPid parameters or even better pass an `Options` struct encoding these parameters and will be easier to expand in the future